### PR TITLE
Finish TypeScript migration

### PR DESCRIPTION
**Pull request by elliotfklein**
### Status
**READY**

### Description
* Fix a few changes that were lost between #2 and #3
* Remove old unused `components/*.js` files
* Convert rest of `.js` files to `.tsx`
* Add linting for Next + React hooks

Still not actually checking types with `strict` mode, [made another branch here](https://github.com/V1Michigan/v1/tree/ts-strict) so we can start working on those errors (not sure it's a priority though).

### Related PRs
Follow-up to #3 

### Dependencies
- ADDED `@typescript-eslint/eslint-plugin`
- ADDED `eslint-config-next`

### Migrations
NO

### Steps to Test or Reproduce

```sh
git pull --prune
git checkout ts-migration-part-2
yarn install
npm run dev
npm run lint:check
```

### CODE REVIEW

**Comment by mdvsh** [comment]
@elliotfklein Looks like the calendar fetch here
https://github.com/V1Michigan/v1/blob/c933a0335769d717cdbecbbebfedda031c2c3ddb/components/calendarapi.tsx#L7
gives a 404 and Next displays JSON parse error in Firefox.

File: components/Account.tsx [hunk 1]
```diff
@@ -1,55 +1,52 @@
...
+      try {
+        setLoading(true);
 
-      const { data, error, status } = await supabase
-        .from("users")
-        .select("username, website, avatar_url")
-        .eq("id", user.id)
-        .single();
+        const { data, error, status } = await supabase
+          .from("users")
```
Comment (mdvsh) [inline components/Account.tsx:17]: The current Account tries to fetch info from `public.users` table, however we have our user info mapped into `public.profiles`. Changing `from` to `profiles` fixes the alert we get on incorrectly querying (checked locally)

**Review by mdvsh — CHANGES_REQUESTED** [review]
Tested updating user as well. Mutations sync with the database. CRU works 🥳 Thank you Elliot!

**Comment by elliotfklein** [comment]
> @elliotfklein Looks like the calendar fetch here
> 
> https://github.com/V1Michigan/v1/blob/c933a0335769d717cdbecbbebfedda031c2c3ddb/components/calendarapi.tsx#L7
> 
> 
> gives a 404 and Next displays JSON parse error in Firefox.
> We might want to update the API endpoint.

@shamdasani who owns the Railway project for this? Don't see it under v1umich@gmail.com

File: components/Account.tsx (hunk 1 above)
Comment (elliotfklein) [reply components/Account.tsx]: Fixed, thanks! (Sorry, I think you mentioned this earlier too, totally forgot)

**Review by mdvsh — APPROVED** [review]
Yup, that should do it! 🚄

**Pull request merged**